### PR TITLE
#164798250 Users should be able to  register.

### PR DESCRIPTION
**Pull request by anyatibrian**
### What does this PR do?

- Creates a functionality that enables a user to register and activate their account

Description of tasks to be completed.

- Add user registration functionality
- Add registration functionality tests
- Add registration modal which takes inputs
- Add error and success messages

How should this be manually tested?
- follow the Heroku link below
 ```http://ah-frontend-prime-pr-14.herokuapp.com/register```,
- enter your ```username, email``` and ```password``` 

Any background context you want to provide?

- N/A

What are the relevant pivotal tracker stories?

[#164798250](https://www.pivotaltracker.com/story/show/164798250)

Screenshots(if appropriate)

login Validation errors 
Screenshot 2019-05-10 at 06 46 21

login Unsucessfull

Screenshot 2019-05-10 at 07 12 16

### CODE REVIEW

**Review by e-ian — APPROVED** [review]
Tested. Good work with frontend validations @anyatibrian. LGTM.

**Review by e-ian — COMMENTED** [review]
@anyatibrian please ensure the branch is deployed to heroku

**Review by patrickf949 — APPROVED** [review]
Tested. LGTM

File: src/tests/components/registerComponent.test.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  });
+  it("should render with fail", () => {
+    expect(wrapper).toMatchSnapshot();
+  });
+  it("should simulate  onchange", () => {
+    const onChange = wrapper.dive();
+    onChange.find("#username").simulate("change");
+    expect(props.handleOnChange).toHaveBeenCalled();
+  });
+  it("should simulate have been clicked", () => {
```
Comment (MuhanguziDavid) [inline src/tests/components/registerComponent.test.js]: have more descriptive test names

File: src/tests/components/registerComponent.test.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    const onSubmit = wrapper.dive();
+    onSubmit.find(".button-success").simulate("click");
```
Comment (MuhanguziDavid) [inline src/tests/components/registerComponent.test.js]: Every test should have an expectation or assertion (this applies for every other test without an `except` statement)

**Review by MuhanguziDavid — CHANGES_REQUESTED** [review]
Nice work @anyatibrian , please
- correct your PR title
- have a descriptive PR description especially `How to test this PR`
- improve the warning message `wrong password must at least one number, lowercase char,special char` when someone enters a password that does not fulfil criteria.
- ensure that error toasts are red
- error toast shouldn’t be timed, a user should see the error and close it themselves.
- Add a loader
- redirect a user to either the login page or the home page after registration.
- fix all eslint issues
- add a screenshot(s) to your Pr description. 
- work on coverage, it could be better than that

**Review by MuhanguziDavid — CHANGES_REQUESTED** [review]
- The login functionality seems to have been broken. It returns an error!?
- Ensure that the register page is responsive
- I see two toasts per action. Ensure to have only one
- Error toasts should not be timed. Just add a statement to remove previous toasts when a new action is performed (@e-ian did it on his task, he could help out).
- ensure that your registration form fulfils the requirements given by the PO e.g. remove the word "enter" from the titles, etc
- punctuate the message on successful registration. It is a bit confusing without punctuation.

**Comment by anyatibrian** [comment]
> Nice work @anyatibrian , please
> 
> * correct your PR title
> * have a descriptive PR description especially `How to test this PR`
> * improve the warning message `wrong password must at least one number, lowercase char,special char` when someone enters a password that does not fulfil criteria.
> * ensure that error toasts are red
> * error toast shouldn’t be timed, a user should see the error and close it themselves.
> * Add a loader
> * redirect a user to either the login page or the home page after registration.
> * fix all eslint issues
> * add a screenshot(s) to your Pr description.
> * work on coverage, it could be better than that

done

**Comment by anyatibrian** [comment]
> * The login functionality seems to have been broken. It returns an error!?
> * Ensure that the register page is responsive
> * I see two toasts per action. Ensure to have only one
> * Error toasts should not be timed. Just add a statement to remove previous toasts when a new action is performed (@e-ian did it on his task, he could help out).
> * ensure that your registration form fulfils the requirements given by the PO e.g. remove the word "enter" from the titles, etc
> * punctuate the message on successful registration. It is a bit confusing without punctuation.

all done

**Comment by anyatibrian** [comment]
> * The login functionality seems to have been broken. It returns an error!?
> * Ensure that the register page is responsive
> * I see two toasts per action. Ensure to have only one
> * Error toasts should not be timed. Just add a statement to remove previous toasts when a new action is performed (@e-ian did it on his task, he could help out).
> * ensure that your registration form fulfils the requirements given by the PO e.g. remove the word "enter" from the titles, etc
> * punctuate the message on successful registration. It is a bit confusing without punctuation.

all done

**Review by Genza999 — APPROVED** [review]
Great work @anyatibrian . Kindly fix the spelling error in the 3rd commit message i.e registration and also make the _finishes_ statement start with a capital F.

**Pull request merged**